### PR TITLE
fix repetition of exported tag with $CATEGORYn(category) when several leaves

### LINKED ISSUE #6683: metadata export: $(CATEGORYn(category)) includes multiple copies of level n if there are tags in level n+1
**Describe the bug**
When using the $(CATEGORYn(category)) variable on the "edit metadata exportation", multiple copies of the n-level tag are exported if there are also n+1-level tags attached to the image (one per extra n+1 tag).

**To Reproduce**
- Create the following tag tree:
```
location
  |- Argentina
     |- Bariloche
     |- Rosario
```
- Attach to an image the tags "Argentina" and "Bariloche"
- Define the following mapping on the "edit metadata exportation" dialog:
```
Xmp.photoshop.Country -> $(CATEGORY0(location))
```
- Export the image

The "Country" field in the resulting image gets populated with "Argentina,Argentina"

**Expected behavior**
The "Country" field in the resulting image should be "Argentina" (only one)

**Platform:**
 - Darktable Version: 3.2.1
 - OS: Kubuntu 20.04.1 LTS with KDE backports, up-to-date

**Additional context**
- Removing the "Bariloche" tag results in a single "Argentina" in the export field
- Conversely, attaching both "Bariloche" and "Rosario" results in three copies of "Argentina" in the export field
- I compared old exports from the same image, and this behavior was not present on version 3.0.2 under Kubuntu 19.10 (namely, a single "Argentina" was exported even if "Bariloche" was also attached)
- I suspect of PR #4983, but I can't check it

### PR DESCRIPTION
**Pull request by phweyland**
Fixes #6683

### CODE REVIEW

File: src/common/tags.c [hunk 1]
```diff
@@ -1685,7 +1685,17 @@ char *dt_tag_get_subtags(const gint imgid, const char *category, const int level
     {
       gchar **pch = g_strsplit(tag, "|", -1);
       char *subtag = pch[rootnb + level];
-      tags = dt_util_dstrcat(tags, "%s,", subtag);
+      gboolean valid = TRUE;
+      // check we have not yet this subtag in the list
+      printf("tags %s subtag %s\n", tags, subtag);
```
Comment (TurboGit) [inline src/common/tags.c:1690]: you need to remove this debug printf.

**Comment by phweyland** [comment]
:(,  of course !

**Review by TurboGit — APPROVED** [review]
Thanks!

File: src/common/tags.c [hunk 2]
```diff
@@ -1685,7 +1685,16 @@ char *dt_tag_get_subtags(const gint imgid, const char *category, const int level
     {
       gchar **pch = g_strsplit(tag, "|", -1);
       char *subtag = pch[rootnb + level];
-      tags = dt_util_dstrcat(tags, "%s,", subtag);
+      gboolean valid = TRUE;
+      // check we have not yet this subtag in the list
+        if(tags && strlen(tags) >= strlen(subtag) + 1)
+      {
+        char *found = g_strstr_len(tags, strlen(tags), subtag);
+        if(found[strlen(subtag)] == ',')
```
Comment (parafin) [inline src/common/tags.c:1693]: This line is missing check for found == NULL. It causes a crash reported by torero on IRC:
https://pastebin.com/Jzbk2i4h
https://pastebin.com/0FaLbKMB

**Comment by parafin** [comment]
This PR needs a bug-fix.

**Pull request merged**